app/controllers/api/v1/packages_controller.rb

```diff
@@ -42,12 +44,16 @@ def show
     api :POST, "/v1/packages", "Create a package"
     param_group :package
     def create
-      @package = Package.new(package_params)
-      @package.offer_id = offer_id
-      if @package.save
-        render json: @package, serializer: serializer, status: 201
+      if package_record
+        @package.offer_id = offer_id
+        if @package.save
+          save_item_details
+          render json: @package, serializer: serializer, status: 201
+        else
+          render json: @package.errors.to_json, status: 422
+        end
       else
-        render json: @package.errors.to_json, status: 422
+        render nothing: true, status: 204
```

Comment:
I think this case is where we should create the package normally - as if it had come in from a non-stockit request. i.e. 

      @package = Package.new(package_params)
      @package.offer_id = offer_id
      if @package.save
        render json: @package, serializer: serializer, status: 201
      else
        render json: @package.errors.to_json, status: 422
      end